### PR TITLE
edit/add/delete ticket

### PR DESCRIPTION
**Pull request by i-iurchuk**

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -4,40 +4,165 @@ const middlewares = jsonServer.defaults();
...
-  const tickets = db.tickets.filter(ticket => columns.find(el => el.id === ticket.columnId));
+  const boardId = req.params.id;
   
-  res.jsonp({ ...board, columns, tickets });
-})
+  if(boardId !== null) {
+    const board = db.boards.find(({ id }) => id == boardId);
+
+    if(!!board){
+      const columns = db.columns.filter( column => column.boardId == boardId)
```
Comment (Steinmar) [inline api/index.js:23]: Optional. For preventing types converting in comparing you can always convert types for a number or a string and use strict compare

File: client/src/components/CreateTicketBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+export interface CreateTicketBlockProps {
+  columnId: string;
+  onTicketSave(data: { title: string; columnId: string; }): void;
+}
+
+/*
+* COMPONENT
+*/
+
+export const CreateTicketBlock: React.FC<CreateTicketBlockProps> = ({ columnId, onTicketSave }) => {
```
Comment (Steinmar) [inline client/src/components/CreateTicketBlock.tsx:17]: I'll check this on for correct work with React.memo

File: client/src/components/CreateTicketBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  const classes = useStyles();
+  const [showFullTicket, setShowFullTicket] = useState(true);
+  const [title, setTitle] = useState('');
+  
+  const handleCloseAddTicket = () => setShowFullTicket(!showFullTicket);
```
Comment (Steinmar) [inline client/src/components/CreateTicketBlock.tsx:22]: Please move this realization to one method and use this method in all places where you need to call  setShowFullTicket(!showFullTicket);

File: client/src/pages/BoardPage/BoardPage.tsx [hunk 4]
```diff
@@ -1,46 +1,118 @@
...
+  const classes = useStyles();
+  const [showTicketModal, setShowTicketModal] = useState()
+  
   useEffect(() => {
     fetchBoard(match.params.id);
   }, [match.params.id]);
 
-  const orderedColumns = [...columns].sort((a, b) => a.order - b.order)
+  useEffect(() => {
+    setShowTicketModal(location.pathname.includes('/ticket/'))
```
Comment (Steinmar) [inline client/src/pages/BoardPage/BoardPage.tsx:32]: I'll investigate a possible solution for this issue

File: client/src/pages/TicketPage/TicketPage.tsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  const handleEditableTextBlur = (data) => onUpdate(id, data);
+  const handleDeleteTicket = () => onDelete(id);
+  const handleSelectFiles = ( { target: { files }}) => setFiles([...files]);
+  const handleUploadFiles = () => {
+    uploadTicketResources({id, resources: [...files].map(({ name }) => name)}) // Dirty hack for fake api
+    setFiles([])
+  };
+
+  useEffect(() => {
+    const ticketId = location.pathname.split('/')[4];
```
Comment (Steinmar) [inline client/src/pages/TicketPage/TicketPage.tsx:38]: Please use useLocation hook for working with location.

**Review by Steinmar — COMMENTED** [review]
General. Please use useDispatch and useSelector hooks instead of mapStateToProps since you are working with functional components.

File: api/index.js (hunk 1 above)
Comment (i-iurchuk) [reply api/index.js]: fixed

File: client/src/components/CreateTicketBlock.tsx (hunk 2 above)
Comment (i-iurchuk) [reply client/src/components/CreateTicketBlock.tsx]: fixed

File: client/src/components/CreateTicketBlock.tsx (hunk 3 above)
Comment (i-iurchuk) [reply client/src/components/CreateTicketBlock.tsx]: done

File: client/src/pages/TicketPage/TicketPage.tsx (hunk 5 above)
Comment (i-iurchuk) [reply client/src/pages/TicketPage/TicketPage.tsx:35]: done. thnx!

File: client/src/pages/TicketPage/TicketPage.tsx [hunk 6]
```diff
@@ -1,41 +1,39 @@
...
-  error,
-  location,
+export const TicketPageComponent: React.FC<TicketPageProps> = ({ 
+  onUpdate,
+  onDelete,
 }) => { 
   const classes = useStyles();
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const { id, error, boardId, title, description, resources } = useSelector((state: AppState) => state.ticketPage);
```
Comment (Steinmar) [inline client/src/pages/TicketPage/TicketPage.tsx:23]: Please have a look at reselect

**Pull request merged**